bool req
---
int32[] color
float32[] cup_pos


